docs/src/tutorial/minimal_working_compute.jl

```diff
@@ -0,0 +1,290 @@
+                        -1,
+                    ),
+                ],
+            ),
+        ),
+    ),
+)
+
+# Now make a descriptor pool to allocate the buffer descriptors from (not a big
+# one, just 1 descriptor set with 1 descriptor in total), ...
+dpool = unwrap(create_descriptor_pool(device, 1, [DescriptorPoolSize(DESCRIPTOR_TYPE_STORAGE_BUFFER, 1)]))
+
+# ... allocate the descriptors for our layout, ...
+dsets = unwrap(allocate_descriptor_sets(device, DescriptorSetAllocateInfo(dpool, [dsl])))
+dset = first(dsets)
+
+# ... and make the descriptors point to the right buffers.
+update_descriptor_sets(
+    device,
+    [
+        WriteDescriptorSet(
+            dset,
+            0,
+            0,
+            DESCRIPTOR_TYPE_STORAGE_BUFFER,
+            [],
+            [DescriptorBufferInfo(buffer, 0, WHOLE_SIZE)],
+            [],
+        ),
+    ],
+    [],
+)
+
+# ## Executing the shader
+#
+# Let's create a command pool in the right queue family, and take a command
+# buffer out of that.
+cmdpool = unwrap(create_command_pool(device, qfam_idx))
+cbufs = unwrap(allocate_command_buffers(device, CommandBufferAllocateInfo(cmdpool, COMMAND_BUFFER_LEVEL_PRIMARY, 1)))
+cbuf = first(cbufs)
+
+# Now that we have a command buffer, we can fill it with commands that cause
+# the kernel to be run. Basically, we bind and fill everything, and then
+# dispatch a sufficient amount of invocations of the shader to span over the
+# array.
+begin_command_buffer(cbuf, CommandBufferBeginInfo(flags = COMMAND_BUFFER_USAGE_ONE_TIME_SUBMIT_BIT))
+
+cmd_bind_pipeline(cbuf, PIPELINE_BIND_POINT_COMPUTE, p)
+
+const_buf = [shader_push_consts(1.234, data_items)]
+cmd_push_constants(cbuf, pl, SHADER_STAGE_COMPUTE_BIT, 0, sizeof(shader_push_consts), Ptr{Nothing}(pointer(const_buf)))
+
+cmd_bind_descriptor_sets(cbuf, PIPELINE_BIND_POINT_COMPUTE, pl, 0, [dset], [])
+
+cmd_dispatch(cbuf, div(data_items, const_local_size_x, RoundUp), 1, 1)
+
+end_command_buffer(cbuf)
+
+# Finally, find a handle to the compute queue and send the command to execute
+# the shader!
+compute_q = get_device_queue(device, qfam_idx, 0)
+queue_submit(compute_q, [SubmitInfo([], [], [cbuf], [])])
+
+# ## Getting the data
+#
+# After submitting the queue, the data is being crunched in the background. To
+# get the resulting data, we need to wait for completion and invalidate the
+# mapped memory (so that whatever data updates that happened on the GPU get
+# transferred to the mapped range visible for the host).
+#
+# While [`queue_wait_idle`](@ref) will wait for computations to be carried out,
+# we need to make sure that the required data is kept alive during queue
+# operations. In non-global scopes, such as functions, the compiler may skip
+# the allocation of unused variables or garbage-collect variables that the
+# runtime thinks are no longer used, thus possibly breaking the initialization
+# logic of Vulkan objects. In this particular case, the runtime is not aware
```

Comment:
Tried to make this passage a bit clearer.

```suggestion
# the allocation of unused variables or garbage-collect objects that the
# runtime thinks are no longer used. If garbage-collected, objects will call
# their finalizers which imply the destruction of the Vulkan objects
# (via `vkDestroy...`). In this particular case, the runtime is not aware
```

Reply:
lgtm :]